src/clover/tests/integration/test_clover.py

```diff
@@ -0,0 +1,1727 @@
+                "Average daily stored energy supplied / kWh"
+            ],
+            0.0,
+        )
+        self.assertEqual(info_file_data["simulation_1"]["final_storage_size"], 0.0)
+        self.assertEqual(info_file_data["simulation_1"]["initial_storage_size"], 0.0)
+
+    def test_diesel_grid_no_pv_storage(self):
+        """
+        Tests the case with diesel, grid, PV and storage.
+
+        The test simulation uses:
+            - a storage system size of 25 kWh.
+
+        """
+
+        info_file_data = self._run_clover_simulation(
+            True, True, False, True, storage_size=25
+        )
+
+        # Check appraisal criteria
+        self.assertEqual(
+            info_file_data["simulation_1"]["system_appraisal"]["criteria"]["blackouts"],
+            0.099,
+        )
+        self.assertEqual(
+            info_file_data["simulation_1"]["system_appraisal"]["criteria"][
+                "cumulative_cost"
+            ],
+            15841.411,
+        )
+        self.assertEqual(
+            info_file_data["simulation_1"]["system_appraisal"]["criteria"][
+                "cumulative_ghgs"
+            ],
+            42714.989,
+        )
+        self.assertEqual(
+            info_file_data["simulation_1"]["system_appraisal"]["criteria"]["lcue"],
+            0.628,
+        )
+        self.assertEqual(
+            info_file_data["simulation_1"]["system_appraisal"]["criteria"][
+                "renewables_fraction"
+            ],
+            0.0,
+        )
+        self.assertEqual(
+            info_file_data["simulation_1"]["system_appraisal"]["criteria"][
+                "unmet_energy_fraction"
+            ],
+            0.026,
+        )
+
+        # Check diesel parameters
+        self.assertEqual(
+            info_file_data["simulation_1"]["analysis_results"][
+                "Average daily diesel energy supplied / kWh"
+            ],
+            11.149,
+        )
+        self.assertEqual(
+            info_file_data["simulation_1"]["analysis_results"]["Diesel times"], 0.511
+        )
+        self.assertEqual(info_file_data["simulation_1"]["diesel_capacity"], 3.0)
+
+        # Check grid parameters
+        self.assertEqual(
+            info_file_data["simulation_1"]["analysis_results"][
+                "Average daily grid energy supplied / kWh"
+            ],
+            7.196,
+        )
+        self.assertEqual(
+            info_file_data["simulation_1"]["analysis_results"][
+                "Average grid availability / hours/day"
+            ],
+            9.338,
+        )
+
+        # Check PV parameters
+        self.assertEqual(
+            info_file_data["simulation_1"]["analysis_results"][
+                "Average daily renewables energy used / kWh"
+            ],
+            0.0,
+        )
+        self.assertEqual(
+            info_file_data["simulation_1"]["analysis_results"][
+                "Cumulative pv generation / kWh"
+            ],
+            0.0,
```

Comment:
This if the first place (of 7) within the file where a `no-pv` test is carried out. (You can search for these with `_no_pv_` to find them all.)
In this instance, I believe that the test is probably right: where there is no PV, we want the output to state that there was no PV generation. However, it currently does the calculation based off the renewables.ninja profile. I'm happy to leave as is.. but I think we should probably change it :smile: